plugins/outputs/kinesis/kinesis.go

```diff
@@ -163,6 +163,21 @@ func writekinesis(k *KinesisOutput, r []*kinesis.PutRecordsRequestEntry) time.Du
 	return time.Since(start)
 }
 
+func (k *KinesisOutput) getPartitionKey(metric telegraf.Metric) string {
+	if k.RandomPartitionKey {
+		u := uuid.NewV4()
+		return u.String()
+	}
+	if k.PartitionKey != "" {
```

Comment:
The struct will have the zero value of the type.  For strings it will be "".  I assume this is the reason why the current ordering for assigning the final partitionKey was chosen.